app/forms/api/profile_creation_form.rb

```diff
@@ -0,0 +1,151 @@
+# rubocop:disable IdentityIdp/ErrorsAddLinter
+module Api
+  class ProfileCreationForm
+    include ActiveModel::Model
+
+    validate :valid_user_bundle
+    validate :valid_user
+    validate :valid_password
+
+    attr_reader :user_password, :user_bundle, :user_bundle_headers, :user_bundle_payload
+    attr_reader :user_session
+    attr_reader :pii, :profile, :personal_key
+
+    def initialize(user_password:, user_bundle:, user_session:)
+      @user_password = user_password
+      @user_bundle = user_bundle
+      @user_session = user_session
+      set_idv_session
+    end
+
+    def submit
+      @form_valid = valid?
+
+      if form_valid?
+        create_profile
+        cache_encrypted_pii
+        generate_personal_key
+
+      end
+
+      FormResponse.new(
+        success: form_valid?,
+        errors: errors.to_hash,
+        extra: extra_attributes,
+      )
+    end
+
+    private
+
+    def create_profile
+      profile_maker = build_profile_maker
+      profile = profile_maker.save_profile
+      @pii = profile_maker.pii_attributes
+      @profile = profile
+    end
+
+    def cache_encrypted_pii
+      cacher = Pii::Cacher.new(current_user, session)
+      cacher.save(user_password, profile)
+    end
+
+    def generate_personal_key
+      @personal_key = profile.encrypt_recovery_pii(pii)
+    end
+
+    def complete_session
+      complete_profile if phone_confirmed?
+      create_gpo_entry if session[:address_verification_mechanism] == 'gpo'
+    end
+
+    def phone_confirmed?
+      session[:vendor_phone_confirmation] == true && session[:user_phone_confirmation] == true
+    end
+
+    def complete_profile
+      current_user.pending_profile&.activate
+      move_pii_to_user_session
+    end
+
+    def create_gpo_entry
+      move_pii_to_user_session
+      self.pii = Pii::Attributes.new_from_json(user_session[:decrypted_pii]) if pii.is_a?(String)
+      confirmation_maker = GpoConfirmationMaker.new(
+        pii: pii, service_provider: service_provider,
+        profile: profile
```

Comment:
Minor: Can we be consistent with line breaks?

```suggestion
        pii: pii,
        service_provider: service_provider,
        profile: profile
```